include/solvers/signed_distance_transformation.h

```diff
@@ -0,0 +1,164 @@
+   * modulating the interface thickness.
+   */
+  SignedDistanceTransformationTanh(const double tanh_thickness)
+    : tanh_thickness(tanh_thickness)
+  {}
+
+  /**
+   * @brief Computes the value of the phase fraction from the signed distance.
+   *
+   * @param[in] signed_distance Value of the signed distance
+   *
+   * @return Value of the computed phase fraction after applying the
+   * transformation.
+   */
+  double
+  transfom_signed_distance(const double signed_distance) override
+  {
+    return 0.5 - 0.5 * tanh(signed_distance / tanh_thickness);
+  }
+
+private:
+  /**
+   * User-defined parameter that defines the thickness of the interface. It
+   * influences its sharpness: a smaller value leads to a sharper interface.
+   */
+  const double tanh_thickness;
+};
+
+/**
+ * @brief Transforms the signed distance to the phase fraction with a piecewise
+ * polynomial function (degree 4).
+ *
+ * The phase fraction is defined as:
+ *
+ * if \f$d < 0 \f$
+ * \f$\phi = 0.5 - 0.5(4d' + 6d'^2 + 4*d'^3 + d'^4)\f$
+ *
+ * else
+ * \f$\phi = 0.5 - 0.5(4d' - 6d'^2 + 4*d'^3 - d'^4)\f$
+ *
+ * with \f$d' = d/d_\mathrm{max}\f$, where \f$d\f$ the signed distance and
+ * \f$d_\mathrm{max}\f$ the maximum redistanciation distance. This
+ * transformation clamps the phase fraction to 0 or 1 when \f$d = \pm
+ * d_\mathrm{max}\f$.
+ *
+ */
+class SignedDistanceTransformationPiecewisePolynomial
+  : public SignedDistanceTransformationBase
+{
+public:
+  /**
+   * @brief Constructor of the piecewise polynomial signed distance
+   * transformation.
+   *
+   * @param[in] max_reinitialization_distance Value of the maximum distance
+   */
+  SignedDistanceTransformationPiecewisePolynomial(
+    const double max_reinitialization_distance)
+    : max_reinitialization_distance_inv(1.0 / max_reinitialization_distance)
+  {}
+
+  /**
+   * @brief Computes the value of the phase fraction from the signed distance.
+   *
+   * @param[in] signed_distance Value of the signed distance
+   *
+   * @return Value of the computed phase fraction after applying the transtion.
+   */
+  double
+  transfom_signed_distance(const double signed_distance) override
+  {
+    const double dimentionless_d =
+      signed_distance * max_reinitialization_distance_inv;
+
+    double piecewise_polynomial_value;
+
+    if (dimentionless_d < 0.0)
+      {
+        piecewise_polynomial_value =
+          4.0 * dimentionless_d +
+          6.0 * Utilities::fixed_power<2>(dimentionless_d) +
+          4.0 * Utilities::fixed_power<3>(dimentionless_d) +
+          Utilities::fixed_power<4>(dimentionless_d);
+      }
+    else
+      {
+        piecewise_polynomial_value =
+          4.0 * dimentionless_d -
+          6.0 * Utilities::fixed_power<2>(dimentionless_d) +
+          4.0 * Utilities::fixed_power<3>(dimentionless_d) -
+          Utilities::fixed_power<4>(dimentionless_d);
+      }
+    return 0.5 - 0.5 * piecewise_polynomial_value;
```

Comment:
I think we could do something like this to avoid calculating the powers twice (recheck the signs, I am not 100% sure)

```suggestion
    dimentionless_d = abs(dimentionless_d);
    piecewise_polynomial_value = 4.0 * dimentionless_d -
          6.0 * Utilities::fixed_power<2>(dimentionless_d) +
          4.0 * Utilities::fixed_power<3>(dimentionless_d) -
          Utilities::fixed_power<4>(dimentionless_d);
      
    if (signed_distance < 0.0)
      {
         return 0.5 + 0.5 * piecewise_polynomial_value;
      }
    else
      {
        return 0.5 - 0.5 * piecewise_polynomial_value;
      }
```